lib/eclint.ts

```diff
@@ -179,18 +196,29 @@ module eclint {
 						}
 						try {
 							if (rule.type === 'DocumentRule') {
-								(<DocumentRule>rule).check(context, settings, doc);
+								(<DocumentRule>rule).check(settings, doc).forEach(addError);
 							} else {
 								var check = (<LineRule>rule).check;
 								doc.lines.forEach(line => {
-									check(context, settings, line);
+									addError(check(settings, line));
 								});
 							}
 						} catch (err) {
 							done(createPluginError(err));
 						}
 					});
 
+					updateResult(file, {
+						// Forced type conversion, not wrapper class
+						fixed: Boolean(file.editorconfig && file.editorconfig.fixed),
```

Comment:
Have you considered `!!A` in cases like these?
```ts
fixed: !!(file.editorconfig && file.editorconfig.fixed),
```
Also, you can use `lodash/get` here like so:
```ts
import { get } from 'lodash';
fixed: !!get(file, 'editorconfig.fixed'),
```